src/pages/beheer/materiaalaanvragen/index.tsx

```diff
@@ -121,6 +129,21 @@ const CPMaterialRequestsPage: NextPage<DefaultSeoInfo> = ({ url }) => {
 	const renderEmptyMessage = (): ReactNode =>
 		tHtml('pages/beheer/materiaalaanvragen/index___geen-materiaal-aanvragen');
 
+	const renderDetailBlade = () => {
+		return (
+			<MaterialRequestDetailBlade
+				isOpen={!isLoading && isDetailBladeOpen}
+				onClose={() => setIsDetailBladeOpen(false)}
+				currentMaterialRequestDetail={currentMaterialRequestDetail}
+			/>
+		);
+	};
+
+	const onRowClick = (evt: MouseEvent<HTMLTableRowElement>, row: Row<MaterialRequest>) => {
+		setCurrentMaterialRequest(row.original);
+		setIsDetailBladeOpen(true);
```

Comment:
personal preference: this saves state twice.
You can use the setCurrentMaterialRequest to know if the modal should be open. If a material request is selected => open the model, if currentMaterialRequest is null => close the modal

```
const [currentMaterialRequest, setCurrentMaterialRequest] = useState<MaterialRequest | null>(null);
```

Saving state twice sometimes can cause strange behavior if these things get out of sync. eg: if currentMaterialRequest is not set but the isDetailBaldeOpen is set to true.

Reply:
I'm getting issues with the scrolllock and can't seem to fix this when using 1 state..